server/server.go

```diff
@@ -19,64 +19,82 @@ package server
 import (
 	"context"
 	"net/http"
-	"os"
-	"os/signal"
-	"syscall"
 	"time"
 
 	"github.com/Peripli/service-manager/rest"
-	"github.com/Sirupsen/logrus"
 	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
 )
 
-// Configuration represents the configuration to use for the server
-type Configuration interface {
-	// Address returns the port on which the server to listen. The port must be preceded by a colon, for example ":8080"
-	Address() string
-
-	// RequestTimeout returns the read/write request timeout
-	RequestTimeout() time.Duration
-
-	// ShutdownTimeout returns the timeout for which the server to wait running tasks before terminating
-	ShutdownTimeout() time.Duration
-}
-
 // Server is the server to process incoming HTTP requests
 type Server struct {
-	Configuration Configuration
+	Configuration *Config
 	Router        *mux.Router
 }
 
 // New creates a new server with the provided REST API configuration and server configuration
 // Returns the new server and an error if creation was not successful
-func New(api rest.API, config Configuration) (*Server, error) {
+func New(api rest.API, config *Config) (*Server, error) {
 	router := mux.NewRouter().StrictSlash(true)
 	registerControllers(router, api.Controllers())
 
-	return &Server{config, router}, nil
+	setUpLogging(config.LogLevel, config.LogFormat)
+
+	return &Server{
+		Configuration: config,
+		Router:        router,
+	}, nil
 }
 
 // Run starts the server awaiting for incoming requests
-func (server *Server) Run() {
+func (server *Server) Run(ctx context.Context) {
 	handler := &http.Server{
 		Handler:      server.Router,
-		Addr:         server.Configuration.Address(),
-		WriteTimeout: server.Configuration.RequestTimeout(),
-		ReadTimeout:  server.Configuration.RequestTimeout(),
+		Addr:         server.Configuration.Address,
+		WriteTimeout: server.Configuration.RequestTimeout,
+		ReadTimeout:  server.Configuration.RequestTimeout,
 	}
-	startServer(handler, server.Configuration.ShutdownTimeout())
+	startServer(handler, server.Configuration.ShutdownTimeout, ctx)
+}
+
+func moveRoutes(prefix string, fromRouter *mux.Router, toRouter *mux.Router) error {
```

Comment:
Couldn't find better way to embed another router's handlers at a path-prefix.

Reply:
maybe we can pick a less-weirdly sounding name